hashintel/hash
# Rename `Knowledge*` to `Persisted*`

## 🌟 What is the purpose of this PR?
As we migrate the workspace we've introduced some new types that live alongside the older definitions (to minimise diffs). For the ontology elements we've used the prefix `Persisted*` (for example `PersistedEntityType`) as these represent _persisted_ records, and therefore have the associated metadata such as `ownedById`, etc.

We had also made parallel types for "knowledge" types such as `Entity`, for these we used the prefix `Knowledge*` to avoid conflicts.

This PR renames `Knowledge*` to `Persisted*` for consistency.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1200211978612931/1203089535761796/f) _(internal)_

## 🔍 What does this change?

See commits

## 📜 Does this require a change to the docs?

- I don't believe these are mentioned anywhere. As we restructure the types we should improve documentation though to better explain the temporary state and such.

## ⚠️ Known issues

Loads of types are inconsistent

## 🐾 Next steps

- Rework GraphQL types to cover the subgraphs, etc.

## 🛡 What tests cover this?

- Manual testing of the type-editor route
- The "ontology" tests in `integration`
- The "knowledge" tests in `integration`

## ❓ How to test this?

1.  Checkout the branch / view the deployment
2. Ensure you take down the docker deployment and clear all volumes.
3. Run `external-services up --build` in the repo root
4. Inside `packages/hash/integration` in another terminal run
  a.  `LOG_LEVEL="debug" yarn jest "ontology" --maxWorkers 1`  
  b.  `LOG_LEVEL="debug" yarn jest "knowledge" --maxWorkers 1`
5. Also try `yarn dev` and check that login works, and that the `/type-editor` route is not broken.

## 📹 Demo
<img width="780" alt="image" src="https://user-images.githubusercontent.com/25749103/193837334-a6f466d0-5760-400b-adc2-dfe6760b1897.png">
<img width="547" alt="image" src="https://user-images.githubusercontent.com/25749103/193837655-2719050c-874c-4b9b-8264-193164f36ef1.png">
